### ISSUE TITLE
`./lib/flatten.cpp:1241: unsigned int MiniZinc::EnvI::registerArrayEnum(const std::vector<unsigned int>&): Assertion 'i <= _enumVarDecls.size() || i <= _tupleTypes.size() || i <= _recordTypes.size()' failed.`

### ISSUE THREAD

**Issue by LebedevRI** [MiniZinc/libminizinc#833]
```
include "connected.mzn";

int: NUM_GRAPH_NODES = 3;
int: NUM_GRAPH_EDGES = length(GRAPH_EDGES);

enum GRAPH_NODES = GN(1..NUM_GRAPH_NODES);
enum GRAPH_EDGES = GE(1..length(j,i in GRAPH_NODES where i > j)(1));

array[GRAPH_EDGES] of tuple(GRAPH_NODES,GRAPH_NODES): GRAPH_NODE_PAIRS = [ (j,i) | j,i in GRAPH_NODES where i > j ];
array[GRAPH_EDGES] of GRAPH_NODES: GRAPH_LEAVING_NODE  = [ p.1 | p in GRAPH_NODE_PAIRS ];
array[GRAPH_EDGES] of GRAPH_NODES: GRAPH_ENTERING_NODE = [ p.2 | p in GRAPH_NODE_PAIRS ];

array[GRAPH_NODES] of var bool: GraphNodes;
array[GRAPH_EDGES] of var bool: GraphEdges;

array[GRAPH_NODES,GRAPH_NODES] of var bool: ReachabilitySetNodesMatrix;
array[GRAPH_NODES,GRAPH_EDGES] of var bool: ReachabilitySetEdgesMatrix;

constraint (
  forall (n in GRAPH_NODES)(
    connected(
        GRAPH_LEAVING_NODE,
        GRAPH_ENTERING_NODE,
        ReachabilitySetNodesMatrix[n,..],
        row(ReachabilitySetEdgesMatrix, n)
%         ReachabilitySetEdgesMatrix[n,..]
      )
    )
);
```
```
minizinc: ./lib/flatten.cpp:1241: unsigned int MiniZinc::EnvI::registerArrayEnum(const std::vector<unsigned int>&): Assertion `i <= _enumVarDecls.size() || i <= _tupleTypes.size() || i <= _recordTypes.size()' failed.
Process finished with non-zero exit code 6.
```

Is no fuzzing being run on the codebase?
While i'm very good at writing broken code,
i'm a bit surprized to be encountering _so_ many failing assertions.

**Comment by cyderize** [comment]
This one seems to have already been fixed in 2.8.5 (or maybe 2.8.4) and in the current `develop` branch (at least when running normally - not sure if you're running with other options enabled).

**Comment by LebedevRI** [comment]
Sorry for the noise :(
I've filled https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1079024.
